src/components/app/header/index.tsx

```diff
@@ -1,82 +1,324 @@
+'use client'
```

Comment:
we could get more fine-grain where we `use client` by loading `viewer` data from the layout RSC but also using this component in the pages router means we need to `useViewer` hook top level to do viewer related checks